ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -50,7 +63,7 @@ public KsqlContext() {
    *
    * @param streamsProperties
    */
-  public KsqlContext(Map<String, Object> streamsProperties) {
+  private KsqlContext(Map<String, Object> streamsProperties) {
```

Comment:
put the logic in here into the `create(Map)` method and have it call the new constructor below.

Reply:
Done!